### PR TITLE
Datagen key file support and test scaffolding

### PR DESCRIPTION
**Pull request by robertbastian**
This is adapted from #1460

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]
I like the choice of leaving in the test scaffolding for key-extract so it's easy to use this test later too

File: tools/datagen/README.md [hunk 1]
```diff
@@ -11,39 +11,39 @@ More details on each tool can be found by running `--help`.
```
Comment (sffc) [inline tools/datagen/README.md:11]: Nit: Here and in main.rs: Please add

```
3. `icu4x-key-extract`: Extract `ResourceKey` objects present in a compiled executable.
```

or remove all references to icu4x-keyextract until we actually add that tool

File: tools/scripts/data.toml [hunk 2]
```diff
@@ -48,21 +60,6 @@ args = [
     "--overwrite",
 ]
 
-[tasks.testdata-build-hello-world]
```
Comment (sffc) [inline tools/scripts/data.toml:51]: Issue: This is used to re-generate hello_world.postcard.

Please restore this task, and optionally add more comments or documentation either here or in a new file like provider/blob/tests/README.md.

**Review by sffc — CHANGES_REQUESTED** [review]
I cross-referenced this with #1460.  I noticed one thing missing and another thing that shouldn't have been deleted.

File: tools/scripts/data.toml (hunk 2 above)
Comment (robertbastian) [reply tools/scripts/data.toml:51]: Ah this was showing as removed in #1460 because it hasn't been synced in a while.

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -453,7 +458,7 @@ jobs:
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
         # keep args in sync with `clippy-all` in .cargo/config.toml
-        args: --all-targets --all-features -- -D warnings
+        args: --all-targets --all-features -- -D warnings -Aclippy::unknown-clippy-lints -Aclippy::field-reassign-with-default
```
Comment (sffc) [inline .github/workflows/build-test.yml:461]: Issue: Please revert this change.  It may have originated from a messy merge if you pulled in one of my commits and then rebased it to a newer version.

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -759,10 +764,8 @@ jobs:
         rustup component add rust-src
         rustup toolchain list
         rustup toolchain install nightly-2021-12-22
-        cd ffi/diplomat/wasm
-        npm ci
-        echo "$HOME/work/icu4x/icu4x/ffi/diplomat/wasm/node_modules/wasm-opt/bin" >> $GITHUB_PATH
-        echo "$HOME/work/icu4x/icu4x/ffi/diplomat/wasm/node_modules/wabt/bin" >> $GITHUB_PATH
+        sudo npm install -g wasm-opt --unsafe-perm
+        sudo npm install -g wabt
```
Comment (sffc) [inline .github/workflows/build-test.yml:768]: Issue: Please revert this change.

**Review by sffc — APPROVED** [review]
I'm approving but please fix the messy diffs in build-test.yml.  Only one change is required: adding

```
      - name: Build Examples and test icu4x-key-extract
        uses: actions-rs/cargo@v1.0.1
        with:
          command: make
          args: wasm-compare-worklog-keys
```

My PR also moved the Install Node.js up in the process, but this is unnecessary and could be done in a different, smaller PR.
